crates/krator/src/manifest.rs

```diff
@@ -1,20 +1,30 @@
 use core::pin::Pin;
 use core::task::{Context, Poll};
 use tokio::sync::watch::{channel, Receiver, Sender};
-use tokio_stream::Stream;
+use tokio_stream::{wrappers::WatchStream, Stream};
 
 /// Wrapper for `ObjectState::Manifest` type which reflects
 /// the latest version of the object's manifest.
-#[derive(Clone)]
-pub struct Manifest<T: Clone> {
+pub struct Manifest<T: Clone + Sync + Send + std::marker::Unpin + 'static> {
     rx: Receiver<T>,
+    stream: WatchStream<T>,
 }
 
-impl<T: Clone> Manifest<T> {
+impl<T: Clone + Sync + Send + std::marker::Unpin + 'static> Clone for Manifest<T> {
+    fn clone(self: &Manifest<T>) -> Manifest<T> {
+        Manifest {
+            rx: self.rx.clone(),
+            stream: WatchStream::new(self.rx.clone()),
```

Comment:
I know this looks super hacky, but there does not appear to be a way to get the `latest()` value from a `WatchStream` because the `Receiver` is stored in a private field. `Receiver` should be pretty lightweight to clone, and calling `borrow()` does not appear to update the `version` stored, so there shouldn't be any issue with these two being "out of sync".

Reply:
Can you add this as a comment in the code? I think this is useful context to have down the road